Build error from glog in PixSfM /src

Installation went smoothly for me until the final pixSfM package install via pip. Compilation reached 40%, then returned this error from the glog source packaged with pixSfM:

`/pixel-perfect-sfm/pixsfm/util/src/glog.cc:86:48: error: invalid conversion from ‘void (*)(const char*, int)’ to ‘void (*)(const char*, size_t)’ {aka ‘void (*)(const char*, long unsigned int)’} [-fpermissive]
86 |            []() { google::InstallFailureWriter(&PyBindLogStack); })`

The full error context is reprinted below:

```
[ 40%] Building CXX object pixsfm/CMakeFiles/pixsfm.dir/bundle_adjustment/src/bundle_adjustment_options.cc.o
/home/kevin/pixel-perfect-sfm/pixsfm/util/src/glog.cc: In lambda function:
/home/kevin/pixel-perfect-sfm/pixsfm/util/src/glog.cc:86:48: error: invalid conversion from ‘void (*)(const char*, int)’ to ‘void (*)(const char*, size_t)’ {aka ‘void (*)(const char*, long unsigned int)’} [-fpermissive]
86 |            []() { google::InstallFailureWriter(&PyBindLogStack); })
|                                                ^~~~~~~~~~~~~~~
|                                                |
|                                                void (*)(const char*, int)
In file included from /usr/local/include/colmap/util/logging.h:37,
from /home/kevin/pixel-perfect-sfm/pixsfm/util/src/glog.cc:1:
/usr/local/include/glog/logging.h:1972:12: note:   initializing argument 1of ‘void google::InstallFailureWriter(void (*)(const char*, size_t))’
1972 |     void (*writer)(const char* data, size_t size));
|     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [pixsfm/CMakeFiles/pixsfm.dir/build.make:132: pixsfm/CMakeFiles/pixsfm.dir/util/src/glog.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:368: pixsfm/CMakeFiles/pixsfm.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/home/kevin/pixel-perfect-sfm/setup.py", line 112, in <module>
setup(
File "/home/kevin/anaconda3/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
return distutils.core.setup(**attrs)
File "/home/kevin/anaconda3/lib/python3.9/distutils/core.py", line 148, in setup
dist.run_commands()
File "/home/kevin/anaconda3/lib/python3.9/distutils/dist.py", line 966, in run_commands
self.run_command(cmd)
File "/home/kevin/anaconda3/lib/python3.9/distutils/dist.py", line 985, in run_command
cmd_obj.run()
File "/home/kevin/anaconda3/lib/python3.9/site-packages/setuptools/command/develop.py", line 34, in run
self.install_for_development()
File "/home/kevin/anaconda3/lib/python3.9/site-packages/setuptools/command/develop.py", line 114, in install_for_development
self.run_command('build_ext')
File "/home/kevin/anaconda3/lib/python3.9/distutils/cmd.py", line 313, in run_command
self.distribution.run_command(command)
File "/home/kevin/anaconda3/lib/python3.9/distutils/dist.py", line 985, in run_command
cmd_obj.run()
File "/home/kevin/pixel-perfect-sfm/setup.py", line 55, in run
self.build_extension(ext)
File "/home/kevin/pixel-perfect-sfm/setup.py", line 98, in build_extension
subprocess.check_call(['cmake', '--build', '.'] + build_args,
File "/home/kevin/anaconda3/lib/python3.9/subprocess.py", line 373, in check_call
raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j']' returned non-zero exit status 2.
----------------------------------------
ERROR: Command errored out with exit status 1: /home/kevin/anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/kevin/pixel-perfect-sfm/setup.py'"'"'; __file__='"'"'/home/kevin/pixel-perfect-sfm/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
```

Thanks @Skydes for your fast, helpful reply.

To answer your question, I did not compile glog from scratch, but rather through the usual package (on Arch), e.g.:
`sudo pacman -Syu google-glog`

For the error I referenced, I made your proposed substitution (`size` --> `size_t`), but it looks like I'll have the patch elsewhere as other related variable conversion errors are halting my build, for example:

```
[  3%] Building CXX object pixsfm/CMakeFiles/pixsfm.dir/util/src/glog.cc.o
/home/kevin/pixel-perfect-sfm/pixsfm/util/src/glog.cc: In lambda function:
/home/kevin/pixel-perfect-sfm/pixsfm/util/src/glog.cc:86:48: error: invalid conversion from ‘void (*)(const char*, int)’ to ‘void (*)(const char*, size_t)’ {aka ‘void (*)(const char*, long unsigned int)’} [-fpermissive]
86 |            []() { google::InstallFailureWriter(&PyBindLogStack); })
|                                                ^~~~~~~~~~~~~~~
|                                                |
|                                                void (*)(const char*, int)
In file included from /usr/local/include/colmap/util/logging.h:37,
from /home/kevin/pixel-perfect-sfm/pixsfm/util/src/glog.cc:1:
/usr/local/include/glog/logging.h:1972:12: note:   initializing argument 1of ‘void google::InstallFailureWriter(void (*)(const char*, size_t))’
1972 |     void (*writer)(const char* data, size_t size));
```

--- a/pixsfm/util/src/glog.cc
+++ b/pixsfm/util/src/glog.cc
@@ -16,7 +16,10 @@ namespace py = pybind11;
 class glog_dummy {};  // dummy class
 
 namespace pixsfm {
-void PyBindLogStack(const char* data, int size) {
+
+// Issue #7: Glog version > 0.5.0 requires T=size_t, <= 0.5.0 T=int
+template <typename T>
+void PyBindLogStack(const char* data, T size) {
   std::chrono::milliseconds timespan(2000);  // or whatever
   py::scoped_estream_redirect stream(
       std::cerr,                                // std::ostream&

